cli/bzk/service.go

```diff
@@ -0,0 +1,414 @@
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		err = ensureContainerIsStarted(client, getWebRunOptions(config.Registry, config.Tag))
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+func restartService(cmd *cli.Cmd) {
+	cmd.Action = doRestartService
+}
+
+func doRestartService() {
+	client, err := docker.NewDocker("")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	allContainers, err = client.Ps(&docker.PsOptions{
+		All: true,
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	config, err := getConfig()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = ensureContainerIsStarted(client, getMongoRunOptions())
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = restartContainer(client, getServerRunOptions(config.Registry, config.Home, config.DockerSock, config.SCMKey, config.Tag))
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	err = restartContainer(client, getWebRunOptions(config.Registry, config.Tag))
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func upgradeService(cmd *cli.Cmd) {
+	cmd.Action = func() {
+		log.Printf("Pulling Bazooka images to check for new versions. This may take some time...\n")
+
+		client, err := docker.NewDocker("")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		bzkImages, err := client.Images(&docker.ImagesOptions{})
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		var hasError bool
+		for _, image := range bzkImages {
+			for _, tag := range image.RepoTags {
+				if strings.HasPrefix(tag, "bazooka/") {
+					err = client.Pull(&docker.PullOptions{Image: tag})
+					if err != nil {
+						log.Printf("Unable to pull image %s, reason is: %v\n", tag, err)
+						hasError = true
+					}
+					log.Printf("Newest image %s upgraded\n", tag)
+					break
+				}
+			}
+		}
+		if hasError {
+			log.Fatalf("Error while pulling some images, see above errors\n")
+		}
+
+		log.Printf("Bazooka Images have been upgraded, let's restart bazooka\n")
+		doRestartService()
+	}
+}
+
+func stopService(cmd *cli.Cmd) {
+	cmd.Action = func() {
+		client, err := docker.NewDocker("")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		allContainers, err = client.Ps(&docker.PsOptions{
+			All: true,
+		})
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		err = stopContainer(client, "bzk_mongodb")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		err = stopContainer(client, "bzk_server")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		err = stopContainer(client, "bzk_web")
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+}
+
+func statusService(cmd *cli.Cmd) {
+	cmd.Action = func() {
+		client, err := docker.NewDocker("")
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		allContainers, err = client.Ps(&docker.PsOptions{
+			All: true,
+		})
+		if err != nil {
+			log.Fatal(err)
+		}
+
+		mongoUp := getContainerStatus("bzk_mongodb")
+		serverUp := getContainerStatus("bzk_server")
+		webUp := getContainerStatus("bzk_web")
+
+		if mongoUp && serverUp && webUp {
+			log.Printf("Bazooka service is Up\n")
```

Comment:
Do we need the date/time for this ? wouldn't a simple `fmt.Printf` be better suited ?